### DISCUSSION TITLE
Mermaid formatter/linter?

### DISCUSSION THREAD

**Discussion by Yokozuna59** [mermaid-js/mermaid#4362] [Ideas]
Hi there! What an amazing tool!

But I've got a question: is there any tool for formatting or linting mermaid code? For both `.md` and `.mmd` files.

For example, following some best practices or unwritten rules for making diagrams consistent and so on. Or is there any rule for each diagram to make it better.

**Comment by Yokozuna59** [comment]
It appears that there isn't one yet.

I'm planning to build one here: [Yokozuna59/mermaid-lint](https://github.com/Yokozuna59/mermaid-lint). Anyone is welcome to contribute and help me! Even thoughts would be fine.

But since there isn't an AST yet (#2523), I'll create it using [langium](https://github.com/langium/langium). See #4401 for more information.

**Reply by Yokozuna59** [reply]
Update: The package will be merged in mermaid repo in `packages/parser`.

The PR: #4450

**Comment by jasonworden** [comment]
Just made https://github.com/jasonworden/mermaid-lint :)

Also see http://jasonworden.com/mermaid-lint
